File: android/core/model/src/main/java/com/creapolis/solennix/core/model/EventFormLink.kt
package com.creapolis.solennix.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class EventFormLink(
    val id: String = "",
    @SerialName("user_id") val userId: String = "",
    val token: String = "",
    val label: String? = null,
    val status: String = "active",
    @SerialName("submitted_event_id") val submittedEventId: String? = null,
    @SerialName("submitted_client_id") val submittedClientId: String? = null,
    val url: String = "",
    @SerialName("expires_at") val expiresAt: String = "",
    @SerialName("used_at") val usedAt: String? = null,
    @SerialName("created_at") val createdAt: String = "",
    @SerialName("updated_at") val updatedAt: String = ""
)

@Serializable
data class GenerateEventFormLinkRequest(
    val label: String? = null,
    @SerialName("ttl_days") val ttlDays: Int? = null
)

File: ios/Packages/SolennixCore/Sources/SolennixCore/Models/EventFormLink.swift
import Foundation

public struct EventFormLink: Codable, Identifiable, Sendable {
    public let id: String
    public let userId: String
    public let token: String
    public var label: String?
    public let status: String
    public var submittedEventId: String?
    public var submittedClientId: String?
    public let url: String
    public let expiresAt: String
    public var usedAt: String?
    public let createdAt: String
    public let updatedAt: String

    public init(id: String, userId: String, token: String, label: String? = nil,
                status: String, submittedEventId: String? = nil, submittedClientId: String? = nil,
                url: String, expiresAt: String, usedAt: String? = nil,
                createdAt: String, updatedAt: String) {
        self.id = id
        self.userId = userId
        self.token = token
        self.label = label
        self.status = status
        self.submittedEventId = submittedEventId
        self.submittedClientId = submittedClientId
        self.url = url
        self.expiresAt = expiresAt
        self.usedAt = usedAt
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}
